crates/bevy_utils/src/label.rs

```diff
@@ -47,6 +50,39 @@ where
     }
 }
 
+/// Trait for implementors of `*Label` types that support downcasting.
+pub trait LabelDowncast {
+    // FIXME: use "return position impl Trait in traits" when that stabilizes.
+    /// The type returned from [`downcast_from`](#method.downcast_from).
+    type Output: Deref<Target = Self>;
+    /// Attempts to convert data from a label to type `Self`. Returns a reference-like type.
+    fn downcast_from(data: u64) -> Option<Self::Output>;
+}
+
+#[doc(hidden)]
+pub struct VTable {
+    // FIXME: When const TypeId stabilizes, inline the type instead of using a fn pointer for indirection.
```

Comment:
This appears to be stabilized? https://github.com/rust-lang/rust/pull/72488

Reply:
It got reverted in https://github.com/rust-lang/rust/pull/77083, unfortunately.

Reply:
```suggestion
    // FIXME: When const TypeId stabilizes, inline the type instead of using a fn pointer for indirection.
    // See https://github.com/rust-lang/rust/issues/77125
```